packages/pass-style/src/byteArray.js

```diff
@@ -71,3 +73,82 @@ export const ByteArrayHelper = harden({
       );
   },
 });
+harden(ByteArrayHelper);
+
+// /////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Returns a Uint8Array reflecting the current contents of `byteArray`.
+ *
+ * @param {ArrayBuffer} byteArray i.e., a hardened Immutable ArrayBuffer whose
+ * `passStyle` is `'byteArray'`. NOTE: A normal (mutable) ArrayBuffer is also
+ * accepted, in which case the returned `Uint8Array` is a non-freezeable
+ * read-write view of `byteArray`.
+ * (Due to a special case in `harden`, the returned Uint8Array, even if
+ * permanently non-freezable, can still be hardened.)
+ * @returns {Uint8Array} a view into the contents of `byteArray`.
+ */
+export const byteArrayToUint8Array = byteArray => {
+  if (getPrototypeOf(byteArray) === ArrayBuffer.prototype) {
+    // If `byteArray` inherits directly from `ArrayBuffer.prototype`,
+    // then assume `byteArray` is a genuine `ArrayBuffer`,
+    // as it would be in a native implementation
+    // of the Immutable ArrayBuffer proposal. This conversion to a Uint8Array
+    // is then zero-copy. If `byteArray` is indeed an Immutable ArrayBuffer,
+    // then the returned Uint8Array is freezable.
+    return new Uint8Array(byteArray);
+  } else {
+    // Otherwise, assume `byteArray` may be a shim-emulated
+    // Immutable ArrayBuffer. This `.slice()` would then cost a copy into a
+    // fresh mutable normal ArrayBuffer, and the returned Uint8Array would
+    // be a read-write window into that mutable ArrayBuffer.
+    const genuineArrayBuffer = byteArray.slice();
+    return byteArrayToUint8Array(genuineArrayBuffer);
+  }
+};
+harden(byteArrayToUint8Array);
+
+/**
+ * Converts a `Uint8Array` to a `ByteArray`, i.e., a hardened
+ * Immutable ArrayBuffer whose `passStyleOf` is `'byteArray'`.
+ *
+ * @param {Uint8Array} uint8Array
+ * @returns {ArrayBuffer} A ByteArray, i.e., a hardened Immutable ArrayBuffer
+ */
+export const uint8ArrayToByteArray = uint8Array =>
+  // If we're using a native implementation of the full Immutable ArrayBuffer
+  // proposal, and if `uint8Array`'s backing store is such a genuine
+  // Immutable ArrayBuffer, and if the native implementation does the obvious
+  // optimization of `immutableArrayBuffer.sliceToImmutable(0, length)`,
+  // then this conversion is zero-copy.
+  // Otherwise, it pays the copy cost implicit in `sliceToImmutable`.
+  //
+  // @ts-expect-error How can shim add to `ArrayBuffer` ts type?
+  harden(uint8Array.buffer.sliceToImmutable(0, uint8Array.length));
```

Comment:
I think this should account for non-zero buffer offsets (e.g., we expect the ArrayBuffer instance returned from `uint8ArrayToByteArray(new Uint8Array(Uint8Array.of(0, 1, 2, 3, 4, 5, 6).buffer, 1, 4))` to have data [0x01, 0x02, 0x03, 0x04]).
```suggestion
export const uint8ArrayToByteArray = uint8Array => {
  const { byteOffset, length } = uint8Array;
  // If we're using a native implementation of the full Immutable ArrayBuffer
  // proposal, and if `uint8Array`'s backing store is such a genuine
  // Immutable ArrayBuffer, and if the native implementation does the obvious
  // optimization of a non-expanding `sliceToImmutable`, then this conversion is
  // zero-copy. Otherwise, it pays the copy cost implicit in `sliceToImmutable`.
  //
  // @ts-expect-error How can the shim update the `ArrayBuffer` TS type?
  const byteArray = uint8Array.buffer.sliceToImmutable(
    byteOffset,
    byteOffset + length,
  );
  return harden(byteArray);
};
```